### ISSUE TITLE
refactor(backend): move metric-compat watchlist behind BackendAdapter abstraction (HIGH-2 follow-up)

### ISSUE THREAD

**Issue by nbx-liz** [nbx-liz/LizyStudio#403]
## Summary

`_workspace_metric_compatibility_errors` (added in PR-C2 / #399 / [`src/lizystudio/services/workspace.py`](src/lizystudio/services/workspace.py)) hardcodes lizyml-specific metric names (`mape`, `rmsle`, `r2`) at the **Service** layer. This is an architectural drift from the `BackendAdapter` Protocol abstraction (BLUEPRINT §3, `docs/coupling-analysis.md`): Service-level code should not know which metric names a particular backend uses.

## Why this is technical debt

- LizyStudio's design intent (BLUEPRINT §3) is that the Service layer is **backend-agnostic** and delegates per-backend logic to `BackendAdapter`.
- Today the metric watchlist `{"mape", "rmsle", "r2"}` is implicitly tied to lizyml's regression metric vocabulary. A second backend (sklearn `Pipeline`, XGBoost native, etc.) might use different names — the Service-layer guard would silently skip the new backend's incompatible metrics.
- The warning text also references **lizyml 0.11.0's** sMAPE / WAPE in the suggested_fix, which couples Service-layer prose to a specific upstream version.

## Proposed refactor

Add a method to `BackendAdapter` Protocol:

```python
def get_incompatible_metrics(
    self,
    target_series: pd.Series,
    metric_names: set[str],
) -> list[IncompatibleMetric]:
    """Return advisory entries for metrics whose preconditions the
    target column violates. Each entry carries a suggested_fix string
    that may reference backend-specific replacements."""
```

`Service.validate_config` then calls `ws.backend.get_incompatible_metrics(...)` and converts the returned list into the existing `severity="warning"` envelope. The lizyml adapter implementation owns the mape/rmsle/r2 watchlist and the sMAPE/WAPE suggestion text; future backends own their own watchlists.

## Acceptance criteria

- [ ] Add `get_incompatible_metrics` to `BackendAdapter` Protocol (`src/lizystudio/backends/base.py`).
- [ ] Implement on `LizyMLAdapter` mirroring the current logic.
- [ ] Default Protocol method returns `[]` (no-op) so existing tests pass without per-backend updates.
- [ ] `_workspace_metric_compatibility_errors` becomes a thin envelope adapter.
- [ ] All 9 cases in `tests/contract/test_validate_metric_compatibility.py` continue to pass.
- [ ] No regression in `tests/contract/test_fit_tune_severity_filter.py`.
- [ ] HISTORY.md Proposal P-0099 (Change Gate — adding a Protocol method).

## References

- Surfaced during **v0.4.1 pre-release code review** (`code-reviewer` agent HIGH-2 finding, 2026-05-05).
- `docs/coupling-analysis.md` (Adapter abstraction goals).
- BLUEPRINT.md §3 (backend abstraction).
- PR #399 (PR-C2, introduced the validator).
- PR #400 (PR-D1, fit/tune severity gating + task type guard).
- Sibling: a second ML backend implementation is currently a **strategic** roadmap item (ROADMAP.md §3.3) — this refactor unblocks it.

## Priority

**tier-3** — not blocking any user-facing feature, but should land before a second backend ships. No urgency, but earlier is cheaper than later.

## Out of scope

- Adding a second backend implementation (separate work item).
- Refactoring `_workspace_split_errors` to the same pattern (it is currently sklearn-row-count specific, but n_rows is a universal precondition; arguably belongs in Service layer).

**Comment by nbx-liz** [comment]
## v0.6 defer (post-v0.4.1 review)

After the v0.4.1 ship and the v0.5 R-1 prep work landed (PR #406 P-0100 / P-0101, PR #408 P-0099 invariants), this refactor is intentionally deferred to v0.6 alongside any second `BackendAdapter` implementation.

**Rationale**:

1. **Cost vs. value**: the watchlist (`mape` / `rmsle` / `r2`) is genuinely lizyml-specific in its current form. Extracting it behind a `BackendAdapter` method (`metric_compatibility_check(target, metrics) -> list[ValidationError]`) requires designing the abstraction to fit a second backend that doesn't exist yet. Doing it now means designing for a hypothetical caller — exactly the over-abstraction that `~/.claude/CLAUDE.md` "no premature abstraction" guidance warns against.

2. **Symmetry with #403's sibling work**: the closest other lizyml-specific surface is `_workspace_split_errors` (n_splits > n_rows). Both deserve the same treatment, and both should be lifted into the adapter together once a second backend candidate is selected.

3. **No regression risk**: the watchlist behaviour is locked by `tests/contract/test_validate_metric_compatibility.py` (15 cases after PR #407) and `tests/contract/test_fit_tune_severity_filter.py`. Refactoring later cannot silently degrade either.

**Triggers to revisit**:

- A second `BackendAdapter` implementation is being scoped (sklearn `Pipeline` direct, xgboost native, or other) — file the BackendAdapter abstraction PR in the same sprint
- A user-reported bug in the watchlist that exposes a backend-coupling assumption
- HISTORY P-0099 R-1 work uncovers a similar pattern in services/training that needs the same lift

Until then, leaving the watchlist in `services/workspace.py:_workspace_metric_compatibility_errors` per P-0101 Decision is the simpler choice.

**Tracking**: this Issue stays open as a v0.6 candidate. ROADMAP §5 already reflects it as `tier-3` defer.

**Comment by nbx-liz** [comment]
Done in PR #491 (`refactor(backend): move metric-compat watchlist behind BackendCore (#403, P-0106)`). The lizyml regression-metric watchlist (mape/rmsle/r2), the task=regression gate, and the sMAPE/WAPE suggestion prose now live in `LizyMLAdapter.get_incompatible_metrics` behind `BackendCore.get_incompatible_metrics`; `_workspace_metric_compatibility_errors` is a thin envelope. HISTORY.md P-0106 Decision recorded. Note: this also obsoletes #452's `_workspace_metric_compatibility_errors` sub-PR.
